Gh 1908 parquet store readme is out of date (#2120)
gchq/Gaffer: store-implementation/parquet-store/README.md

See the License for the specific language governing permissions and
limitations under the License.

# Parquet Store

Contents:
1. [Notice](#notice)
2. [Introduction](#introduction)
3. [Design overview](#design-overview)
4. [Use cases](#use-cases)
5. [Parquet version](#parquet-version)
6. [Properties file](#properties-file)
4. [Schema](#schema)
5. [Inserting data](#inserting-data)
6. [Queries](#queries)

> 7. [Custom serialisers](##Writing a custom serialiser)

Reviewer: ```suggestion
7. [Custom serialisers](#custom-serialisers)
```

> 7. [Optimisations](README.md#optimisations)

Reviewer: ```suggestion
7. [Optimisations](#optimisations)
```

## Parquet version

As many of the operations on the Parquet store use Spark to implement the partitioning and sorting logic, the version of Parquet used is determined by the version that Spark uses. Gaffer currently uses Spark version 2.3.2 which has a dependency on Parquet 1.8.3. Parquet 1.11.0 contains upgrades to the version of parquet-format which improve the performance of "indexed" queries against Parquet files - this should increase the performance of random access queries against the Parquet store. Future work should investigate this further, although this may have to wait until Spark has updated the version of Parquet it uses.

## Properties file

The `ParquetStoreProperties` class contains all properties relating to the configuration of the `ParquetStore`. It can be created from a properties file. It has sensible defaults for all parameters but users may want to tune these properties to suit their data. The following properties can be set:

- `spark.master`: The string that sets what mode to run Spark in. By default, if Spark is installed on the machine it will use Spark's defaults, otherwise it will run in local mode using all available threads;

> - `parquet.data.dir`: The directory to use to save the graph;

Reviewer: ```suggestion
- `parquet.data.dir`: The directory used to save the graph;
```

      .build();
```

> To get the best performance you should allow Gaffer to detect the best serialiser or provide a serialiser class that implements `ParquetSerialisation`. See the section on serialisers in this README for further information about serialisers optimised to work with the Parquet store.

Reviewer: ```suggestion
To get the best performance you should allow Gaffer to detect the best serialiser or provide a serialiser class that implements `ParquetSerialisation`. See the section on [serialisers](custom-serialisers) for further information about serialisers optimised to work with the Parquet store.
```
